src/rfc5322.rs

```diff
@@ -325,26 +324,26 @@ impl<'s> Rfc5322Parser<'s> {
     }
 
     /// Consume LWSP (Linear whitespace)
-    #[unstable]
+    /// [unstable]
     pub fn consume_linear_whitespace(&mut self) {
         self.consume_while(|c| { c == '\t' || c == ' ' });
     }
 
     /// Consume a single character from the input.
     #[inline]
-    #[unstable]
+    /// [unstable]
     pub fn consume_char(&mut self) -> char {
         if self.eof() { 
             // TODO: Consider making this return an Option<char>
             panic!("Consuming beyond end of input");
         }
-        let ch_range = self.s.char_range_at(self.pos);
-        self.pos = ch_range.next;
-        ch_range.ch
+        let c = self.s.char_at(self.pos);
```

Comment:
`char_at` is also unstable, but the phrasing of the error message suggests it's more likely to become stable -- char_range_at, however, seems to be discouraged.